providers/okta/connection/connection.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package connection
+
+import (
+	"context"
+	"errors"
+
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/providers-sdk/v1/vault"
+)
+
+type OktaConnection struct {
+	id    uint32
+	Conf  *inventory.Config
+	asset *inventory.Asset
+	// custom connection fields
+	organization string
+	client       *okta.Client
+	token        string
+}
+
+func NewOktaConnection(id uint32, asset *inventory.Asset, conf *inventory.Config) (*OktaConnection, error) {
+	conn := &OktaConnection{
+		Conf:  conf,
+		id:    id,
+		asset: asset,
+	}
+
+	// initialize connection
+	if conf.Type != "okta" {
+		return nil, errors.New("provider type does not match") // TODO: switch to plugin.ErrProviderTypeDoesNotMatch
+	}
+
+	if conf.Options == nil || conf.Options["organization"] == "" {
+		return nil, errors.New("okta provider requires an organization id. please set option `organization`")
+	}
+
+	org := conf.Options["organization"]
+
+	// if a secret was provided, it always overrides the env variable since it has precedence
```

Comment:
To me, this comment was a bit misleading as the actual handling of env var or token flag happens somewhere else.

Reply:
Great catch, definitely a copy error. The new model is more clear and this issue does not apply here anymore. I am going to remove this comment